### PR TITLE
`azurerm_sql_database` - support for the `extended_auditing_policy` property

### PR DESCRIPTION
**Pull request by yupwei68**
Similar to https://github.com/terraform-providers/terraform-provider-azurerm/pull/5036
continuation of https://github.com/terraform-providers/terraform-provider-azurerm/pull/4982
sql db auditing pass

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
hi @yupwei68

Thanks for this PR.

Since there's still review comments from #4982 pending (which I'm assuming you're working through) I'm going to hold off reviewing this for the moment - once those have been addressed we'll take another look through :)

Thanks!

**Comment by yupwei68** [comment]
Hi @tombuildsstuff , sorry to block you, because I think that the closed PR won't be reviewed any more.  I have replied comments in https://github.com/terraform-providers/terraform-provider-azurerm/pull/4982. Please review.

**Comment by yupwei68** [comment]
hi @tombuildsstuff , any problem?

**Comment by yupwei68** [comment]
@tombuildsstuff @katbyte I've removed some less useful fields. Please continue reviewing.

**Comment by tombuildsstuff** [comment]
@yupwei68 looks like this is still waiting on the changes from #4982?

**Comment by yupwei68** [comment]
@tombuildsstuff I have made some updates. Please continue reviewing it and https://github.com/terraform-providers/terraform-provider-azurerm/pull/5036

**Comment by yupwei68** [comment]
Hi @katbyte , auditing block in resource_arm_sql_database has been updated according to changes in [PR resource_arm_sql_server](https://github.com/terraform-providers/terraform-provider-azurerm/pull/5036). Please help to review.

File: azurerm/internal/services/sql/tests/resource_arm_sql_database_test.go [hunk 1]
```diff
@@ -418,6 +418,36 @@ func TestAccAzureRMSqlDatabase_bacpac(t *testing.T) {
...
 
+func TestAccAzureRMSqlDatabase_updateWithBlobAuditingPolices(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_sql_database", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMSqlDatabaseDestroy,
+		Steps: []resource.TestStep{
+			{
```
Comment (katbyte) [inline azurerm/internal/services/sql/tests/resource_arm_sql_database_test.go]: Could we place a basic step first to make sure we can update to enable

File: azurerm/internal/services/sql/tests/resource_arm_sql_database_test.go [hunk 2]
```diff
@@ -418,6 +418,36 @@ func TestAccAzureRMSqlDatabase_bacpac(t *testing.T) {
...
+				Config: testAccAzureRMSqlDatabase_withBlobAuditingPolicesUpdated(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSqlDatabaseExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "extended_auditing_policy.0.storage_account_access_key_is_secondary", "false"),
+					resource.TestCheckResourceAttr(data.ResourceName, "extended_auditing_policy.0.retention_in_days", "11"),
+				),
+			},
+			data.ImportStep("administrator_login_password", "extended_auditing_policy.0.storage_account_access_key", "create_mode"),
+		},
```
Comment (katbyte) [inline azurerm/internal/services/sql/tests/resource_arm_sql_database_test.go]: and then a basic step at the end to make sure we can disable them?

File: azurerm/internal/services/sql/tests/resource_arm_sql_database_test.go [hunk 3]
```diff
@@ -418,6 +418,36 @@ func TestAccAzureRMSqlDatabase_bacpac(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMSqlDatabase_updateWithBlobAuditingPolices(t *testing.T) {
```
Comment (katbyte) [inline azurerm/internal/services/sql/tests/resource_arm_sql_database_test.go]: COuld we add a test with a single step config `testAccAzureRMSqlDatabase_withBlobAuditingPolices`

**Pull request merged**